src/main/java/life/qbic/datamodel/identifiers/SampleCodeFunctions.java

```diff
@@ -61,7 +63,7 @@ public static boolean isQbicBarcode(String code) {
    * @return true if String is a QBiC entity code, false if not
    */
   public static boolean isQbicEntityCode(String code) {
-    Pattern entityPattern = Pattern.compile("Q[A-X0-9]{4}ENTITY-[1-9][0-9]*", Pattern.CASE_INSENSITIVE);
```

Comment:
This pattern is different from the new one. I think the new one is correct but wanted to note that here.

Reply:
It is an entity id pattern (not a sample id!) and is correct.